gravshort-tree.c

```diff
@@ -257,16 +257,19 @@ int force_treeev_shortrange(TreeWalkQueryGravShort * input,
                         otherh = nop->u.d.hmax;
                 }
                 if(h < otherh)
-                {
                     h = otherh;
-                    if(r2 < h * h)
-                    {
-                        if(nop->f.MixedSofteningsInNode)
-                        {
-                            no = nop->u.d.nextnode;
 
-                            continue;
-                        }
+                /* We want to open the node if it is partially but not entirely
+                 * within the softening length of the particle.
+                 * Note this differs from vanilla Gadget, which
+                 * only opens nodes if they have a larger
+                 * softening than the particle.
+                 * This should reduce force coupling.*/
+                if(r2 < h * h)
+                {
+                    if(r2 > (h + nop->len/2)*(h + nop->len/2)) {
```

Comment:
(h + nop->len/2)**2 > h2 so this will never be hit.

Reply:
Ah indeed, I screwed up my conditionals. Duh.